src/goto-synthesizer/goto_synthesizer_parse_options.cpp

```diff
@@ -47,23 +48,64 @@ int goto_synthesizer_parse_optionst::doit()
   if(result_get_goto_program != CPROVER_EXIT_SUCCESS)
     return result_get_goto_program;
 
-  // TODO
-  // Migrate synthesizer and tests from goto-instrument to goto-synthesizer
+  // Synthesize loop invariants and annotate them into `goto_model`
+  enumerative_loop_contracts_synthesizert synthesizer(goto_model, log);
+  const auto &invariant_map = synthesizer.synthesize_all();
+  const auto &assigns_map = synthesizer.get_assigns_map();
 
+  if(cmdline.isset("dump-loop-contracts"))
   {
-    // Synthesize loop invariants and annotate them into `goto_model`
-    enumerative_loop_contracts_synthesizert synthesizer(goto_model, log);
-    annotate_invariants(synthesizer.synthesize_all(), goto_model);
-    annotate_assigns(synthesizer.get_assigns_map(), goto_model);
-
-    // Apply loop contracts.
-    std::set<std::string> to_exclude_from_nondet_static(
-      cmdline.get_values("nondet-static-exclude").begin(),
-      cmdline.get_values("nondet-static-exclude").end());
-    code_contractst contracts(goto_model, log);
-    contracts.apply_loop_contracts(to_exclude_from_nondet_static);
+    // Default output destination is stdout.
+    std::string json_output_str = "stdout";
```

Comment:
This looked a bit weird to me on first sight, but am I right that this is understood by Crangler?

Reply:
Yes. https://github.com/diffblue/cbmc/blob/develop/src/crangler/c_wrangler.cpp#L616
Alternatively, we can choose not to specify `output`.

Reply:
Would you mind just adding a comment to make sure others also understand this relationship to Crangler.